src/controllers/midi/midicontrollerpresetfilehandler.cpp

```diff
@@ -205,12 +205,12 @@ void MidiControllerPresetFileHandler::addControlsToDocument(const MidiController
     QDomElement controller = doc->documentElement().firstChildElement("controller");
     QDomElement controls = doc->createElement("controls");
 
-    // Iterate over all of the command/control pairs in the input mapping
-    QHashIterator<uint16_t, MidiInputMapping> it(preset.inputMappings);
-    while (it.hasNext()) {
-        it.next();
-
-        const MidiInputMapping& mapping = it.value();
+    // The QHash doesn't guarantee iteration order, so first we sort the keys
```

Comment:
ya, since they are multi-hashes you need to iterate the result of find:
```c++
for (auto it = preset.inputMappings.constFind(key); it.key() == key; ++it) {
  // handle (key, it.value())
}
```

Reply:
oops you also need a check for end:
`it != preset.inputMappings.constEnd() && it.key() == key`

Reply:
that's weird they are just QHash not QMultiHash.  But also, does this affect any other places we've been calling "find"?

Reply:
looks like the other spots iterate over the find correctly